DQMServices/Core/interface/DQMEDHarvester.h

```diff
@@ -1,43 +1,45 @@
 #ifndef CORE_DQMED_HARVESTER_H
 #define CORE_DQMED_HARVESTER_H
 
-//<<<<<< INCLUDES                                                       >>>>>>
-#include "FWCore/Framework/interface/one/EDProducer.h"
 #include "DQMServices/Core/interface/DQMStore.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/one/EDProducer.h"
 
-//<<<<<< PUBLIC DEFINES                                                 >>>>>>
-//<<<<<< PUBLIC CONSTANTS                                               >>>>>>
-//<<<<<< PUBLIC TYPES                                                   >>>>>>
-//<<<<<< PUBLIC VARIABLES                                               >>>>>>
-//<<<<<< PUBLIC FUNCTIONS                                               >>>>>>
-//<<<<<< CLASS DECLARATIONS                                             >>>>>>
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "DataFormats/Histograms/interface/DQMToken.h"
 
-class DQMEDHarvester
-: public edm::one::EDProducer<edm::one::WatchRuns,edm::one::WatchLuminosityBlocks,edm::one::SharedResources,
-edm::EndLuminosityBlockProducer>
+class DQMEDHarvester: public edm::one::EDProducer<edm::Accumulator,
```

Comment:
why do you inherit from `edm::Accumulator`, if you do not read any event data ?

Reply:
The module has to implement either accumulate() or produce(), if none is implemented, it gives a compiler error.

Accumulate makes more sense.

Reply:
My understanding of what the framework will do is:
  - if standard `EDProducer` is not explicitly or implicitly scheduled, its `produce` method will not be called;
  - if an `EDProducer<Accumulator>` is not explicitly or implicitly scheduled, its `accumulate` method will be called for each event.

@Dr15Jones , sorry for bothering you, but did I understand correctly ?

Then, using a standard `EDProducer` for a harvesting module should be (slightly) more efficient.

Reply:
@fwyzard you are correct, the standard produce mechanism would be more efficient.